src/backend_model_instance.h

```diff
@@ -126,10 +126,10 @@ class TritonModelInstance {
   MetricModelReporter* MetricReporter() const { return reporter_.get(); }
 
   // Directly call from C API, so arguments are in the same style
-  void SetMemoryUsage(BufferAttributes** memory_usage, uint32_t usage_size)
+  void SetMemoryUsage(boost::span<BufferAttributes*> memory_usage)
   {
     std::map<TRITONSERVER_MemoryType, std::map<int64_t, size_t>> lusage;
-    for (uint32_t idx = 0; idx < usage_size; ++idx) {
+    for (uint32_t idx = 0; idx < memory_usage.size(); ++idx) {
```

Comment:
You should also be able to do `for (const auto& usage : memory_usage)` with the span so no need for `idx`, if you want to.